devito/passes/iet/openmp.py

```diff
@@ -135,7 +135,20 @@ def _make_clauses(cls, ncollapse=None, chunk_size=None, nthreads=None,
             clauses.append('num_threads(%s)' % nthreads)
 
         if reduction:
-            args = ','.join(str(i) for i in reduction)
+            reds = []
+            for i in reduction:
+                # OMP expects a size not an index as input of reduction, such as
+                # reduction(+:f[1:3])
+                if i.is_Indexed:
+                    bounds = [(k,) if k.is_Number else
+                              (k.subs(d, d.symbolic_min), k.subs(d, d.symbolic_max + 1))
+                              for k, d in zip(i.indices, i.function.dimensions)]
+                    bounds = ','.join(('%s:%s' % v if len(v) > 1 else str(v[0]))
```

Comment:
sorry I'm a bit slow this morning, what does this do?

Reply:
This does `[start:end]` if there is more than one index or just `[index]` if just a single one